lib/src/main/java/grpcbridge/parser/ProtoParser.java

```diff
@@ -0,0 +1,71 @@
+package grpcbridge.parser;
+
+import com.google.protobuf.Message;
+import com.google.protobuf.util.JsonFormat;
+import grpcbridge.http.HttpRequest;
+import grpcbridge.http.HttpResponse;
+import grpcbridge.rpc.RpcMessage;
+import io.grpc.Metadata;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public abstract class ProtoParser implements Parser {
+    private static final String ANY = "*/*";
+    protected abstract String contentType();
+    protected abstract String packMultiple(List<String> serializedItems);
+    public abstract <T extends Message> T parse(@Nullable String body, T.Builder builder);
+
+
+
+    @Override
+    public boolean accept(Collection<String> accepted) {
+        return accepted.stream().anyMatch(it -> supportedTypes().contains(it) || ANY.equals(it));
+    }
+
+    protected List<String> supportedTypes() {
+        return Collections.singletonList(contentType());
+    }
+
+    @Override
+    public RpcMessage parse(HttpRequest httpRequest, Message.Builder builder) {
+        return new RpcMessage(
+            parse(httpRequest.getBody().orElse(null), builder),
+            httpRequest.getHeaders());
+    }
+
+    @Override
+    public HttpResponse serialize(
+        @Nonnull JsonFormat.Printer printer,
+        @Nonnull RpcMessage message
+    ) {
```

Comment:
Join with prev line and make continuation indent 8 chars.

Reply:
done